compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml

```diff
@@ -1,6 +1,7 @@
 -- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
This seems unnecessary.

Reply:
Unfortunately GHC thinks it is. Removing it produces:

```
File:     compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml
Hidden:   no
Range:    320:24-320:76
Source:   typecheck
Severity: DsError
Message: 
  compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:320:25: error:
  • Could not deduce (HasLookupByKey t0 k)
  from the context: HasLookupByKey t k
  bound by the type signature for:
  canSeeContractWithKey : forall t k.
  HasLookupByKey t k =>
  k -> Update Bool
  at compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml:320:25-76
  The type variable ‘t0’ is ambiguous
  • In the ambiguity check for ‘canSeeContractWithKey’
  To defer the ambiguity check to use sites, enable AllowAmbiguousTypes
  In the type signature:
  canSeeContractWithKey : forall t k.
  (HasLookupByKey t k) => k -> Update Bool
ERROR: Compilation failed.
```

Happy to learn of a better way to resolve this if there is one.

Reply:
This is the correct way to solve this. There is no way for GHC to infer the template type since all it has to work with is the key type (two templates can have the same key type) and the return type `Boolean`. This is pretty harmless. Since the introduction of type families you can easily provide the template type using that.

Reply:
I understand why this pragma is necessary in `DA.Internal.Template.Functions`. What I don't understand is why it is necessary here where we don't mention `canSeeContractWithKey` at all.

Reply:
Oh I misunderstood the question. Yeah we can probably kill it here.

Reply:
🤦 

I had put the function in Prelude first, then moved it to Template. Sorry.